### PR TITLE
Support for overlapping SEG segments

### PR DESCRIPTION
**Pull request by Punzo**
Fix for https://github.com/OHIF/Viewers/issues/2164.

generateToolState now would give an array of LabelMaps and segmentsOnFrame. In the case of non-overlapping segments, the arrays will have dimension 1. In case of overlapping segments, the arrays will have dimension M (see https://github.com/dcmjs-org/dcmjs/pull/155/files#diff-077798721d0df020ee8184a2f54d4459dba916e19c86ac0f0aecc4e6ea80bf2fR880) 

This is done, but the UI relative part (OIHF) is still in progress.

### CODE REVIEW

File: src/adapters/Cornerstone/Segmentation_4X.js [hunk 1]
```diff
@@ -669,6 +1000,8 @@ function insertPixelDataPlanar(
             }
         }
     }
+
+    return 1;
```
Comment (swederik) [inline src/adapters/Cornerstone/Segmentation_4X.js:1004]: why is this here?

File: test/test_data.js [hunk 2]
```diff
@@ -227,7 +227,7 @@ const tests = {
       });
   },
 
-  test_multiframe_us: () => {
```
Comment (swederik) [inline test/test_data.js:230]: why is it commented out now? Does it not work?

File: src/adapters/Cornerstone/Segmentation_4X.js [hunk 3]
```diff
@@ -259,9 +261,13 @@ function _createSegFromImages(images, isMultiframe, options) {
...
  * @param  {string[]} imageIds    An array of the imageIds.
  * @param  {ArrayBuffer} arrayBuffer The SEG arrayBuffer.
- * @param {*} metadataProvider
- * @returns {Object}  The toolState and an object from which the
- *                    segment metadata can be derived.
+ * @param  {*} metadataProvider
+ *
+ * @return {[]ArrayBuffer}a list of array buffer for each labelMap
+ * @return {Object} an object from which the segment metadata can be derived
+ * @return {[][][]} 2D list containg the track of segments per frame
```
Comment (swederik) [inline src/adapters/Cornerstone/Segmentation_4X.js:268]: typos for Containing

File: src/adapters/Cornerstone/Segmentation_4X.js [hunk 4]
```diff
@@ -259,9 +261,13 @@ function _createSegFromImages(images, isMultiframe, options) {
...
+ * @return {[][][]} 3D list containg the track of segments per frame for each labelMap
```
Comment (swederik) [inline src/adapters/Cornerstone/Segmentation_4X.js:269]: typo for available

File: src/adapters/Cornerstone/Segmentation_4X.js (hunk 1 above)
Comment (JamesAPetts) [reply src/adapters/Cornerstone/Segmentation_4X.js:1004]: C++ intensifies.

File: test/test_data.js (hunk 2 above)
Comment (Punzo) [reply test/test_data.js:230]: yes it is failing:

(node:39227) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/davide/Development/OIHF/dcmjs/test/cine-test.dcm'

and I don't see any dcmjs/test/cine-test.dcm file in the repo

File: src/adapters/Cornerstone/Segmentation_4X.js (hunk 1 above)
Comment (Punzo) [reply src/adapters/Cornerstone/Segmentation_4X.js:1004]: you are right, the return is not needed

Comment (Punzo) [reply src/adapters/Cornerstone/Segmentation_4X.js:1004]: also https://github.com/dcmjs-org/dcmjs/pull/155/files#diff-077798721d0df020ee8184a2f54d4459dba916e19c86ac0f0aecc4e6ea80bf2fR890 is nto needed. It was something from an old version where I was returning the number of label maps. But now we use the length of the labelmapBufferArray.

Thanks for pointing out

**Comment by Punzo** [comment]
@swederik I squashed the commits, this is ready for merging.
